### ISSUE TITLE
Lots of properties declared as 'private', should be 'protected'

### ISSUE THREAD

**Issue by icefront** [yiisoft/yii2#16532]
### What steps will reproduce the problem?
Override the Controller class, override the getViewPath() function. $this->_viewPath isn't accessible because it's declared as 'private'.

### What is the expected result?
Declare properties which are used by getters/setters as 'protected' to be able to verride the class's getter/setter function without having to override ALL functions that use the property in question.

### What do you get instead?
PHP error

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2, latest
| PHP version      | 5.4
| Operating system | Linux

**Comment by lubosdz** [comment]
I agree with this - framework should enable overriding basically anything in favor of an application.
Framework is a tool -> application is a final product.

**Comment by FabrizioCaldarelli** [comment]
@icefront What are you going to solve making _viewPath protected instead private?

**Comment by rob006** [comment]
`_viewPath` is used only by getter and setter. Whatever you want to do, you can achieve this by overriding this getter and/or setter. You can actually override both and introduce protected `_viewPath` in  your class - parent implementation will be completely ignored then.
